source/Windows.Devices.I2c/I2cControllerManager.cs

```diff
@@ -0,0 +1,48 @@
+﻿//
+// Copyright (c) 2018 The nanoFramework project contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System.Collections;
+
+namespace Windows.Devices.I2c
+{
+    internal sealed class I2cControllerManager
+    {
+        private static readonly object _syncLock = new object();
+
+        // backing field for ControllersCollection
+        // to store the controllers that are open
+        private static Hashtable s_controllersCollection;
+
+        /// <summary>
+        /// <see cref="I2cController"/> collection.
+        /// </summary>
+        /// <remarks>
+        /// This collection is for internal use only.
+        /// </remarks>
+        internal static Hashtable ControllersCollection
+        {
+            get
+            {
+                if (s_controllersCollection == null)
+                {
+                    lock (_syncLock)
+                    {
+                        if (s_controllersCollection == null)
+                        {
+                            s_controllersCollection = new Hashtable();
+                        }
+                    }
+                }
+
+                return s_controllersCollection;
+            }
+
+            set
+            {
+                s_controllersCollection = value;
```

Comment:
If the setter (here and in similar places) gets called, do we have to lock()? Could two threads set this property, or could one set and one read it at the same time?